_posts/2023-07-06-optimize-opensearch-index-shard-size.markdown

```diff
@@ -0,0 +1,109 @@
+Following along with the example, the following API call copies data from the *sample-data-5-1* index to the *sample-data-1-1* index.
+
+```
+POST _reindex
+{
+  "source": {
+    "index": "sample-data-5-1"
+  },
+  "dest": {
+    "index": "sample-data-1-1"
+  }
+}
+```
+
+Note a few additional considerations:
+* Before reindexing or copying data from the old index to the new index you may consider setting the old index to read only. This ensures that no new data is written to the old index during the copy.
+* Free storage space should be considered before reindexing. When copying data to a new index, the new index will consume storage. Ensure you have sufficient free storage before reindexing.
+* When creating a new document, always consider using index templates if available. Index templates contain schema information for an index including data types. Using an index template during index creation helps ensure the correct settings are applied to the new index. Index templates are often used in log analytics workloads where new indexes are created frequently.
+
+Following along with the example, we could run the following API call to block all incoming writes to the *sample-data-5-1* index.
+
+```
+PUT sample-data-5-1/_settings
+{
+  "index": {
+    "blocks.read_only": true
+  }
+}
+```
+
+## Change the number of replica shards
+The number of replicate shards for an index can be easily adjusted after an index is created. The index setting API allows for changing the number of replica shards without needing to create a new index and copying data.
+
+An OpenSearch index has both dynamic and static settings. You can change dynamic index settings at any time, but static settings cannot be changed after index creation. The number of replicas is a dynamic setting, hence why it can be changed without needing to create a new index and copying the data.
+
+Following along with our example, if we want to change the replication factor for the *sample-data-1-1* index from 1 to 0 (that is, no replicates), we can use the index settings API.
+
+```
+PUT sample-data-1-1/_settings
+{
+  "index" : {
+    "number_of_replicas": 0
+  }
+}
+```
+
+## Conclusion and other resources
+Optimizing the number and size of shards in your index can help you have the best performance from OpenSearch. Thinking about and setting the correct number of primary shards when you first create an index is best. However, if your index size changes or you use the default number of primary shards and the defaults don’t work well for your index, you can use this blog as a guide to help you optimize shard counts.
+
+If you prefer to learn about this topic in the format of a video instead of a blog post, check out the YouTube video: [OpenSearch - How to change the number of primary and replica shard(s) of an index](https://www.youtube.com/watch?v=xadv93LlbY4). This blog post is based on the GitHub repository: [OpenSearch_Read_Only_Index](https://github.com/ev2900/OpenSearch_Read_Only_Index)
```

Comment:
```suggestion
If you prefer to learn about this topic in video format instead of a blog post, check out this YouTube video: [OpenSearch - How to change the number of primary and replica shard(s) of an index](https://www.youtube.com/watch?v=xadv93LlbY4). This blog post is based on this GitHub repository: [OpenSearch_Read_Only_Index](https://github.com/ev2900/OpenSearch_Read_Only_Index)
```